knebind/solver/solver.go

```diff
@@ -277,8 +280,22 @@ func (s *solver) solve() (*assign, error) {
 		port2Intfs := make(map[interface{}][]interface{})
 		for dut, node := range dev2Node {
 			for port, intfs := range dev2Node2Port2Intfs[dut.(*opb.Device)][node.(*tpb.Node)] {
+
 				for _, i := range intfs {
-					port2Intfs[port] = append(port2Intfs[port], i)
+					if port.GetGroup() != "" {
+						if i.groupName == port.GetGroup() {
```

Comment:
Oh, I realized that this logic is too strict. We wouldn't be looking for exact equality of the grouping string. We should checking that that groupings match. An Ondatra topology with two ports in group "A" should match against a KNE topology with two ports in group "B". The strings themselves shouldn't matter, right?

Reply:
@greg-dennis No, group name matters lets say we have defined two lags in kne topology group A and B which has two ports each, and we want exactly one port from group A only then we should check for group A's port, for group only there is a strict check but if some one didn't mention about group for a port in ondatra topology, it can come from the grouped ports as well, and for the CLA issue I will check today it self and get back to you

Reply:
But the group name itself doesn't carry any semantic meaning. If I say I want one ports in group "A" and two from group "B", I don't think it matters if I happen to get one port from a group in the KNE topology named "B" and two in the KNE topology named "C" -- as long as the groupings are respected, what does it matter if the same strings are used to represent those groups?

Reply:
@biplamal ^^^

Reply:
@biplamal, yeah, it does make it a lot more complicated to implement, but if these grouping names don't tell us anything besides what ports should stay together in the same group, then I think we'll have to do that.

Reply:
@greg-dennis agreed, that time I thought how we can incorporate grouping without doing any changes in recursive solution finding logic, but now we need do changes which is legit, so I think we should go with some phases
1. Try to understand the existing algorithm and get back to you whether our understanding is correct or not.
2. Propose an algorithm and get reviewed by you.
3.  Then we will implement and raise PR.